### PR TITLE
Import/Export pets

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Implementing a Pair Programming Feature with a text file for #237

### CODE REVIEW

**Comment by Harry-Hopkinson** [comment]
@tonybaloney this pull request should be ready for a review.

File: .eslintrc.json [hunk 1]
```diff
@@ -18,7 +18,6 @@
         "semi": "off",
         "no-unused-vars": "error",
         "unused-imports/no-unused-imports": "error",
-        "prefer-const": "error",
-        "@typescript-eslint/no-non-null-assertion": "warn"
```
Comment (tonybaloney) [inline .eslintrc.json:22]: This warn is really helpful and has caught many bugs

**Comment by tonybaloney** [comment]
File imports will be another blocker to #276 in which browser extensions cannot support the `fs` module.

Please have a look at https://www.npmjs.com/package/vsls with example https://marketplace.visualstudio.com/items?itemName=lostintangent.vsls-whiteboard as APIs for sharing information inside a VS Code live share.

**Comment by Harry-Hopkinson** [comment]
I can use vscode's built in file system. Does that work in the browser?

**Comment by Harry-Hopkinson** [comment]
> File imports will be another blocker to #276 in which browser extensions cannot support the `fs` module.
> 
> Please have a look at https://www.npmjs.com/package/vsls with example https://marketplace.visualstudio.com/items?itemName=lostintangent.vsls-whiteboard as APIs for sharing information inside a VS Code live share.

Does vscode's built in file system work in the browser?

**Comment by tonybaloney** [comment]
I've made a couple of changes:
- The exported files are JSON instead of the text format. The array type was already serializable and now it'll handle the formatting issues
- The input colour is validated to avoid the mystery pet bug
- The exported file uses the "untitled" protocol which opens it in a tab in VS Code ready to save, but without having saved the files so the user can rename it
- The exported file name has  a unique-ish number in it to avoid collision
- The import action uses the open file dialog so the user can pick the file, instead of assuming it's named a certain way and inside the workspace (since the workspace might be a project they're working on and nothing to do with pets).

**Review by tonybaloney — APPROVED** [review]

**Comment by Harry-Hopkinson** [comment]
@tonybaloney thank you

**Pull request merged**